coderd/database/modelmethods.go

```diff
@@ -1,9 +1,36 @@
 package database
 
 import (
+	"golang.org/x/xerrors"
+
 	"github.com/coder/coder/coderd/rbac"
 )
 
+var validAPIKeyScopes = map[string]ApiKeyScope{
+	string(ApiKeyScopeAny):                ApiKeyScopeAny,
+	string(ApiKeyScopeApplicationConnect): ApiKeyScopeApplicationConnect,
+}
+
+func ToAPIKeyScope(v string) (ApiKeyScope, error) {
+	scope, ok := validAPIKeyScopes[v]
+	if !ok {
+		return ApiKeyScope(""), xerrors.Errorf("invalid token scope: %s", v)
+	}
+
+	return scope, nil
+}
```

Comment:
Is this necessary? I could see this being forgotten when adding a value. Postgres will catch this for us as well.